### PR TITLE
frontend/fix: #788 fixed update button in my profile screen

### PR DESCRIPTION
**Pull request by aryanguptaSG**
## Type of Change

- [x] Bugfix
- [ ] New feature
- [ ] Enhancement
- [ ] Refactoring
- [ ] Dependency updates

## Description
fixed update button in my profile screen

### Additional Changes

- [ ] This PR modifies the database schema (database migration added)
- [ ] This PR modifies dhall configs/environment variables
- [ ] This PR contains API breaking changes

## Checklist

- [ ] I formatted the code and addressed linter errors `./dev/format-all-files.sh`
- [ ] I reviewed submitted code
- [ ] I added unit tests for my changes where possible
- [ ] I added a [CHANGELOG](/CHANGELOG.md) entry if applicable

### CODE REVIEW

File: Frontend/ui-customer/src/Screens/CustomerUtils/MyProfileScreen/Controller.purs [hunk 1]
```diff
@@ -128,10 +129,10 @@ eval (UserProfile (GetProfileRes profile)) state = do
...
 eval (EmailIDEditTextAction (PrimaryEditText.TextChanged id value)) state = do
-  if (value == "" && state.data.errorMessage == Just EMAIL_EXISTS) then continue state {props{ isEmailValid = false, isBtnEnabled = false, genderOptionExpanded = false, expandEnabled = false}}
-    else continue state {data {editedEmailId = Just value , errorMessage = if (length value == 0) then Nothing else if ( validateEmail value) then Nothing else Just INVALID_EMAIL },props{ isEmailValid = if (length value == 0) then true else validateEmail value, isBtnEnabled = (length state.data.editedName >=3)  && (if (length value == 0) then true else validateEmail value), genderOptionExpanded = false}}
+  if (value == "" && state.data.errorMessage == Just EMAIL_EXISTS) then continue state {props{isEmailValid = false, isBtnEnabled = false, genderOptionExpanded = (state.props.fromHomeScreen || false), expandEnabled = (state.props.fromHomeScreen || false)}}
```
Comment (Vignesh-772) [inline Frontend/ui-customer/src/Screens/CustomerUtils/MyProfileScreen/Controller.purs:134]: Remove this `|| false` in all cases anyways it is depend on the `state.props.fromHomeScreen`.

File: Frontend/ui-customer/src/Screens/CustomerUtils/MyProfileScreen/Controller.purs [hunk 2]
```diff
@@ -128,10 +129,10 @@ eval (UserProfile (GetProfileRes profile)) state = do
   continue state { data { name = name, editedName = name, gender = gender, emailId = profile.email } }
 eval (NameEditTextAction (PrimaryEditText.TextChanged id value)) state = do
   _ <- pure $ spy "Value changed"  value
-  continue state { data { editedName = value }, props{genderOptionExpanded = false, expandEnabled = false, isBtnEnabled = ((length value >=3) && state.props.isEmailValid )} }
+  continue state { data { editedName = value }, props{genderOptionExpanded = (state.props.fromHomeScreen || false), expandEnabled = (state.props.fromHomeScreen || false), isBtnEnabled = if (state.props.fromHomeScreen) then false else ((length value >=3) && state.props.isEmailValid )} }
```
Comment (Vignesh-772) [inline Frontend/ui-customer/src/Screens/CustomerUtils/MyProfileScreen/Controller.purs:132]: handle `isBtnEnabled` using `&&`

File: Frontend/ui-customer/src/Flow.purs [hunk 3]
```diff
@@ -438,7 +438,7 @@ homeScreenFlow = do
     GO_TO_ABOUT -> aboutUsScreenFlow
     GO_TO_MY_PROFILE  updateProfile -> do
         _ <- pure $ firebaseLogEvent if updateProfile then "safety_banner_clicked" else "ny_user_profile_click"
-        modifyScreenState $ MyProfileScreenStateType (\myProfileScreenState ->  MyProfileScreenData.initData{props{updateProfile = updateProfile , fromHomeScreen = updateProfile}})
+        modifyScreenState $ MyProfileScreenStateType (\myProfileScreenState ->  MyProfileScreenData.initData{props{updateProfile = updateProfile , fromHomeScreen = updateProfile , isBtnEnabled = not updateProfile , genderOptionExpanded = updateProfile , showOptions = true, expandEnabled = true }})
```
Comment (Vignesh-772) [inline Frontend/ui-customer/src/Flow.purs:441]: upload a output video ans squash the commits

**Review by Vignesh-772 — CHANGES_REQUESTED** [review]

File: Frontend/ui-customer/src/Screens/CustomerUtils/MyProfileScreen/Controller.purs (hunk 2 above)
Comment (aryanguptaSG) [reply Frontend/ui-customer/src/Screens/CustomerUtils/MyProfileScreen/Controller.purs]: done

File: Frontend/ui-customer/src/Flow.purs (hunk 3 above)
Comment (aryanguptaSG) [reply Frontend/ui-customer/src/Flow.purs:441]: Output Video Link :- https://drive.google.com/file/d/1-l9zVzEq3s6TtSoQZ4Aq3Sru9Co0Tb8-/view?usp=sharing

**Review by Vignesh-772 — APPROVED** [review]

**Review by witcher-shailesh — APPROVED** [review]

**Pull request merged**